st2client/st2client/models/core.py

```diff
@@ -385,3 +386,95 @@ def re_emit(self, trigger_instance_id, **kwargs):
         return response.json()
+
+
+class PackResourceManager(ResourceManager):
+    @add_auth_token_to_kwargs_from_env
+    def install(self, packs, **kwargs):
+        url = '/%s/install' % (self.resource.get_url_path_name())
+        response = self.client.post(url, {'packs': packs}, **kwargs)
+        if response.status_code != 200:
+            self.handle_error(response)
+        instance = self.resource.deserialize(response.json())
+        return instance
+
+    @add_auth_token_to_kwargs_from_env
+    def remove(self, packs, **kwargs):
+        url = '/%s/uninstall' % (self.resource.get_url_path_name())
+        response = self.client.post(url, {'packs': packs}, **kwargs)
+        if response.status_code != 200:
+            self.handle_error(response)
+        instance = self.resource.deserialize(response.json())
+        return instance
+
+    @add_auth_token_to_kwargs_from_env
+    def create(self, args, **kwargs):
+        url = '/%s/init' % (self.resource.get_url_path_name())
+        keys = ['name', 'description', 'keywords', 'version', 'author',
+                'email']
+        payload = {}
+
+        for key in keys:
+            value = getattr(args, key)
+            if value:
+                payload[key] = value
+
+        response = self.client.post(url, payload, **kwargs)
+        if response.status_code != 200:
+            self.handle_error(response)
+        instance = self.resource.deserialize(response.json())
+        return instance
+
+    @add_auth_token_to_kwargs_from_env
+    def search(self, args, **kwargs):
+        url = '/%s/search' % (self.resource.get_url_path_name())
+        if 'query' in vars(args):
+            payload = {'query': args.query}
+        else:
+            payload = {'pack': args.pack}
+        response = self.client.post(url, payload, **kwargs)
+        if response.status_code != 200:
+            self.handle_error(response)
+        data = response.json()
+        if isinstance(data, list):
+            return [self.resource.deserialize(item) for item in data]
+        else:
+            return self.resource.deserialize(data) if data else None
+
+    @add_auth_token_to_kwargs_from_env
+    def register(self, packs=None, types=None, **kwargs):
+        url = '/%s/register' % (self.resource.get_url_path_name())
+        payload = {}
+        if types:
+            payload['types'] = types
+        if packs:
+            payload['packs'] = packs
+        response = self.client.post(url, payload, **kwargs)
+        if response.status_code != 200:
+            self.handle_error(response)
+        instance = self.resource.deserialize(response.json())
+        return instance
+
+
+class StreamManager(object):
+    def __init__(self, endpoint, cacert, debug):
+        self._url = httpclient.get_url_without_trailing_slash(endpoint) + '/stream'
+        self.debug = debug
+        self.cacert = cacert
+
+    @add_auth_token_to_kwargs_from_env
+    def listen(self, events, **kwargs):
```

Comment:
Can we explicitly declare arguments here? :P

It seems like we should only add token, api_key and then we can get rid of kwargs...